src/pages/CreateAssignment/CreateAssignment.js

```diff
@@ -96,11 +100,63 @@ function CreateAssignment(props) {
     setShowPhonics(false);
     setShowWriting(true);
   }
+  function createLessonCard() {
+    var validCard = true;
+    if (deploymentAccountIds < 1) {
+      setInvalidMessage(invalidMessage => [
+        ...invalidMessage,
+        "Please assign to at least one student."
+      ]);
+      validCard = false;
+    }
+    if (date == null) {
+      setInvalidMessage(invalidMessage => [
+        ...invalidMessage,
+        "Please select a date on the calendar."
+      ]);
+      validCard = false;
+    } else if (date in lessonCards) {
+      setInvalidMessage(invalidMessage => [
+        ...invalidMessage,
+        "Cannot have duplicate dates in the same lesson."
+      ]);
+      validCard = false;
+    } else if (date.getTime() < lessonCreationDate.getTime()) {
+      setInvalidMessage(invalidMessage => [
+        ...invalidMessage,
+        "Cannot select a date before the current date."
+      ]);
+      validCard = false;
```

Comment:
can't you just replace validCards with return; so these functions end early and the validCard conditional is never reached?

Reply:
nah